hydrosdk/modelversion.py

```diff
@@ -654,3 +589,19 @@ def building(self) -> bool:
         :return: if building
         """
         return self.get_status() == ModelStatus.Assembling
+
+    # TODO: Add logging
+    def lock_till_released(self) -> None:
```

Comment:
I suggest taking into consideration this @mkf-simpson [commit](https://github.com/provectus/hydro-auto-od/pull/6/commits/afc7169cb86b9d8dbf3867eebe2646f512402af8) . As I understand, this is done to avoid a situation when an "Assembled" event came before we created `SSEClient`. 

IMO We could add `wait=True` flag into `upload` function to avoid this situation, and create `SSEClient` inside that function before making an API call to upload model tar. Moreover, I guess there is no other use case for `lock_till_released`, and hiding it inside 'upload' will improve overall SDK readability. WDY @trthhrtz @KineticCookie @mkf-simpson ?

Reply:
Generally agree. The problem here is fast failing: if a model fails inside an `upload` request, we'll never recieve any message from SSE. So we have to connect to SSE _before_ `upload` to get `FAILED` event.

Also + for `wait=True` and readability

Reply:
How about now @Valenzione @mkf-simpson @KineticCookie ?

Reply:
Looks much better! 👍